packages/graphql-codegen-cli/src/plugins.ts

```diff
@@ -22,12 +22,12 @@ export async function getPluginByName(
     try {
       return await pluginLoader(moduleName);
     } catch (err) {
-      if (err.code !== 'MODULE_NOT_FOUND' || !err.message.includes(moduleName)) {
+      if (err.code !== 'MODULE_NOT_FOUND') {
```

Comment:
The current plugin/preset loader tries to detect whether the plugin exists or not by requiring it, and catching potential resolution errors that may be thrown. One additional check is made to look for the full require path existing in the message; I believe this was intended to not catch transitive resolution errors occurring within the plugin/preset code itself. However, this isn't compatible with all environments - under PnP for example, we turn the require path into an absolute one in the error message (thus breaking the `indexOf` match).

The current change simply removes this guard; however if you wish to keep it I'm open to implement an alternative (separating the call to `require.resolve` from its `require`, thus being able to only catch errors that come from the plugin/preset resolution itself). It's slightly more involved so I didn't do it yet, let me know what you prefer.